Remove support for Python 3.7 and 3.8

jsf9k opened the pull request:
## 🗣 Description ##

This pull request removes support for Python 3.7 and 3.8.

## 💭 Motivation and context ##

Python 3.7 is [no longer supported](https://www.python.org/downloads/release/python-370/).  [The same is true](https://www.python.org/downloads/release/python-380/) for Python 3.8.

I realize we still have some repositories that must be installed under Python 3.7, but we should make these the exception rather than the rule and explicitly add 3.7 as a supported platform (with a suitable comment) for those.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] Remove the Python 3.7 and 3.8 GH Actions checks.

jsf9k committed:
Remove support for Python 3.7

Python 3.7 is no longer supported:
https://www.python.org/downloads/release/python-370/

dv4harr10 commented:
Hi Team @all , for file setup.py line 85 reads "python_requires=">=3.7", should this be updated to 3.8 or a different value ?

jsf9k committed:
Update python_requires to >=3.8

This is necessary since we have dropped support for Python 3.7.

Co-authored-by: David Harris <123905168+dv4harr10@users.noreply.github.com>

jsf9k commented:
> Hi Team @ALL , for file setup.py line 85 reads "python_requires=">=3.7", should this be updated to 3.8 or a different value ?

Great catch!  See commit d84af0a.

mcdonnnj commented:
@jsf9k If we're dropping support for 3.7 then we should also drop support for 3.8 since it is also end-of-life.

jsf9k committed:
Remove support for Python 3.8

Python 3.8 is no longer supported:
https://www.python.org/downloads/release/python-380/

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

jsf9k commented:
> @jsf9k If we're dropping support for 3.7 then we should also drop support for 3.8 since it is also end-of-life.

Please see commit 9ee2ce9.

dav3r reviewed (approved):
👍
